git-protocol/src/fetch.rs

```diff
@@ -433,32 +433,42 @@ where
     R: AsyncRead + Unpin + Send + 'static,
     W: AsyncWrite + Unpin + Send + 'static,
 {
+    use git_packetline::encode::flush_to_write;
     use git_transport::{
         client::git::{ConnectMode, Connection},
         Protocol,
     };
 
     let stop = AtomicBool::new(false);
-    let conn = Connection::new(
-        recv,
-        send,
-        Protocol::V2,
-        opt.repo.clone(),
-        None::<(String, Option<u16>)>,
-        ConnectMode::Daemon,
-    );
-    let receive_pack = build_receive_pack(&stop);
-    let task = blocking::unblock(move || {
-        let mut delegate = Fetch::new(opt, filter_refs, receive_pack);
-        future::block_on(git_protocol::fetch::fetch(
-            conn,
-            &mut delegate,
-            |_| unreachable!("credentials helper requested"),
-            progress::Discard,
-        ))
-        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-
-        Ok(delegate.out)
+    let task = blocking::unblock({
+        let mut conn = Connection::new(
+            recv,
+            send,
+            Protocol::V2,
+            opt.repo.clone(),
+            None::<(String, Option<u16>)>,
+            ConnectMode::Daemon,
+        );
+        let receive_pack = build_receive_pack(&stop);
+
+        move || {
+            let mut delegate = Fetch::new(opt, filter_refs, receive_pack);
+            future::block_on(async {
+                let res = git_protocol::fetch::fetch(
```

Comment:
Warning: the new and only path for this `git_protocol::fetch(…)`. In future, if there are multiple paths the shorter one should be preferred as [all of these dual-paths](https://github.com/Byron/gitoxide/projects/1#card-65290657) will eventually be cleaned up.